OpenDreamRuntime/Procs/AssignRefProcState.cs

```diff
@@ -0,0 +1,81 @@
+using System.Text;
+using OpenDreamRuntime.Objects;
+
+namespace OpenDreamRuntime.Procs;
+
+internal sealed class AssignRefProcState : ProcState {
+    public static readonly Stack<AssignRefProcState> Pool = new();
+
+    private enum Stage {
+        // Call the associated proc
+        Call,
+
+        // Handle the assignment
+        AssignRef,
+    }
+
+    private DreamObject _dreamObject;
+    private DreamObject? _usr;
+    private readonly DreamValue[] _arguments = new DreamValue[256];
+    private int _argumentCount;
```

Comment:
Shouldn't this only ever need one argument? This could be replaced with a single field.

Reply:
I wasn't really sure why this was done this way in the init object one so I just cargo-culted it. Some operators do take a couple arguments though, notable the output and input ones

Reply:
In InitDreamObject they represent the args passed to `New()`. Since this can be an arbitrary amount the max is fairly large. A small optimization would be to reduce the max amount here to the most you'll ever need for an operator overload.